packages/db/tests/lib/test-granule.js

```diff
@@ -0,0 +1,427 @@
+      execution_cumulus_id: executionCumulusId,
+    }]
+  );
+});
+
+test('upsertGranuleWithExecutionHistory() handles multiple executions for a granule', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    executionPgModel,
+    granulesExecutionsPgModel,
+    collectionCumulusId,
+    executionCumulusId,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'completed',
+  });
+
+  const [granuleCumulusId] = await knex.transaction(
+    (trx) => upsertGranuleWithExecutionHistory(
+      trx,
+      granule,
+      executionCumulusId
+    )
+  );
+
+  const [secondExecutionCumulusId] = await executionPgModel.create(
+    knex,
+    fakeExecutionRecordFactory()
+  );
+
+  await knex.transaction(
+    (trx) => upsertGranuleWithExecutionHistory(
+      trx,
+      granule,
+      secondExecutionCumulusId
+    )
+  );
+
+  const granuleRecord = await granulePgModel.get(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleRecord,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+    }
+  );
+  t.deepEqual(
+    await granulesExecutionsPgModel.search(
+      knex,
+      { granule_cumulus_id: granuleCumulusId }
+    ),
+    [executionCumulusId, secondExecutionCumulusId].map((executionId) => ({
+      granule_cumulus_id: Number(granuleCumulusId),
+      execution_cumulus_id: executionId,
+    }))
+  );
+});
+
+test('upsertGranuleWithExecutionHistory() does not write anything if upserting granule/execution join record fails', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+    granulesExecutionsPgModel,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'running',
+  });
+
+  const fakeGranulesExecutionsPgModel = {
+    upsert: async () => {
+      throw new Error('error');
+    },
+  };
+
+  await t.throwsAsync(
+    knex.transaction(
+      (trx) =>
+        upsertGranuleWithExecutionHistory(
+          trx,
+          granule,
+          executionCumulusId,
+          undefined,
+          fakeGranulesExecutionsPgModel
+        )
+    )
+  );
+
+  t.false(
+    await granulePgModel.exists(
+      knex,
+      {
+        granule_id: granule.granule_id,
+        collection_cumulus_id: collectionCumulusId,
+      }
+    )
+  );
+  t.false(
+    await granulesExecutionsPgModel.exists(
+      knex,
+      {
+        execution_cumulus_id: executionCumulusId,
+      }
+    )
+  );
+});
+
+test('upsertGranuleWithExecutionHistory() will allow a running status to replace a non-running status for different execution', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    executionPgModel,
+    granulesExecutionsPgModel,
+    collectionCumulusId,
+    executionCumulusId,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'completed',
```

Comment:
Nit: 

We added a default to the factory, should we use it here?    Stylistically I'm generally in favor of *explicit* over implicit defaults in testing,  but if we have one.....

Reply:
I added a default to the factory because not in every case where we need a granule fixture do we care about the status. But generally where we do care I set it in the test